ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,152 @@
+	mockSNA.EXPECT().GetLatestSequenceNumber()
+	mockSNA.EXPECT().SetLatestSequenceNumber(testManifest).Return(nil)
+
+	// After we get the latest sequence number, we will expect to set the latest message ID.
+	mockIDA.EXPECT().SetMessageID(testconst.MessageID).Return(nil)
+
+	// Test Metric publication when our responder starts and finishes message processing.
+	mockEntry := mock_metrics.NewMockEntry(ctrl)
+	mockMetrics.EXPECT().New(metrics.TaskManifestHandlingDuration).Return(mockEntry)
+	mockEntry.EXPECT().Done(nil).AnyTimes()
+
+	// When the client queries tasks to be stopped, inject this task to be stopped.
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(testManifest).Return(
+		[]*ecsacs.TaskIdentifier{
+			{DesiredStatus: aws.String("RUNNING"), TaskArn: aws.String(testconst.TaskARN)},
+		},
+		nil,
+	)
+
+	// Handle the task manifest message update.
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	go handleManifestMessage(testManifest)
+
+	// Wait for the ack to send through our channel so we can receive it.
+	taskStopVerifSent := <-stopVerifSent
+
+	// We expect only one task marked for termination, and it should be running.
+	require.Equal(t, len(taskStopVerifSent.StopCandidates), 1)
+	require.Equal(t, aws.StringValue(taskStopVerifSent.StopCandidates[0].TaskArn), testconst.TaskARN)
+}
+
+// TestTaskManifestSequenceNumbers will make sure the manifest can update, retrieve and catch stale
+// Manifest sequence numbers appropriately.
+func TestTaskManifestSequenceNumbers(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	testResponseSender := func(response interface{}) error {
+		return nil
+	}
+
+	mockComparer := mock_session.NewMockTaskComparer(ctrl)
+	mockIDA := mock_session.NewMockManifestMessageIDAccessor(ctrl)
+	mockSNA := mock_session.NewMockSequenceNumberAccessor(ctrl)
+
+	testTaskManifestResponder := NewTaskManifestResponder(mockComparer, mockSNA, mockIDA, metrics.NewNopEntryFactory(), testResponseSender)
+
+	// Create test task manifest.
+	testManifest := setupTestManifestMessage()
+
+	mockSNA.EXPECT().SetLatestSequenceNumber(testManifest).Return(nil).Times(3)
+	testTaskManifestResponder.snAccessor.SetLatestSequenceNumber(testManifest)
+
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(testManifest).Return([]*ecsacs.TaskIdentifier{}, nil)
+
+	// Expect GetLatestSequenceNumber once in handleManifestMessage, and once
+	// when we validate it.
+	mockSNA.EXPECT().GetLatestSequenceNumber().Times(2)
+	mockIDA.EXPECT().SetMessageID(testconst.MessageID).Return(nil)
+
+	// Handle the task manifest message update.
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	handleManifestMessage(testManifest)
+
+	// Now manually try to reset the sequence and make sure it does not update
+	// As 15 <= 15, we should not be able to update to a stale sequence number.
```

Comment:
`testTaskManifestResponder.snAccessor.GetLatestSequenceNumber())` is 0 at this point, is that intentional? Or should we be using `.Return `method on L138 (and L81 for the matter)?